qiskit/pulse/commands/_acquire.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Acquire.
+"""
+
+from qiskit.exceptions import QiskitError
+from qiskit.pulse.commands._pulse_command import PulseCommand
+
+
+class Acquire(PulseCommand):
+    """Acquire."""
+
+    def __init__(self, qubit, memory_slot,
+                 register_slot=None, discriminator=None, kernel=None):
+        """create new acquire command
```

Comment:
That's very good suggestion. As far as I know, `Acquire` class will be stored in a special class of channel group (`AcquireChannel`). Thus, this class could have different behavior from others. However this spec should be discussed carefully. @taalexander can you give us your opinion?

Reply:
I agree with @ajavadia that the qubit should be inferred from the acquire channel that it is applied to. What is less clear to me is how we should handle the `memory_slot` and `register_slot`. Perhaps @ajavadia could clarify how the measure commands currently work within a QuantumCircuit and operating on a `QuantumRegister` and `ClassicalRegister` as this is an analogous situation.

@nkanazawa1989 I think `memory_slot` and `register_slot` should both be optional arguments and we should raise an error if one is not provided. We do not necessarily wish to return the result of every measurement.

Reply:
I removed `qubit` and added `duration` instead. I need to know if it is possible to assign arbitrary register/memory slot for all qubits - or there are any constraints on qubit-register/memory interconnection?

I think `memory_slot` and `register_slot` should be associated with `acquire` command, so that duplication of slot ids can be checked in the scheduler.

Reply:
Why can't the schedule also maintain the `memory_slot` and `register_slot` aspects of the acquire command? Seems similar to `qubit` in that sense.

In circuits, you have a `Measure`, and you're free to apply it to any (qubit, clbit). The circuit will keep the context of where that Measure is applied.

There will be constraints on which qubits can be measured into which registers, and this is reported by the device in the `backend.configuration.register_map`. That will be the job of the scheduler to create a correct schedule based on that information.

Reply:
Because we can save the results of a series of measurements into separate registers, and this could be useful to debug conditionals in the simulator backend. If the constraints on qubit-resgister interconnection exist in the real backend, it would be better to associate `memory_slot` and `register_slot` to the channels, or to make them optional arguments.

Reply:
I think what Ali is suggesting is that there should exist objects `MemorySlot` and `HardwareRegister` (reuse `ClassicalRegister`?) Which would be the input edges to the acquire on the Schedule DAG. These could then be mapped to the appropriate hardware constrained locations during an appropriate Scheduler pass (explicit control could be provided in a similar way to initial_layout).